api/src/routes/protected/module.ts

```diff
@@ -0,0 +1,86 @@
+import { type FastifyPluginCallbackTypebox } from '@fastify/type-provider-typebox';
+
+import { ProgressTimestamp, getPoints } from '../../utils/progress';
+import * as schemas from '../../schemas';
+
+/**
+ * Plugin for the module submission endpoints.
+ *
+ * @param fastify The Fastify instance.
+ * @param _options Options passed to the plugin via `fastify.register(plugin, options)`.
+ * @param done The callback to signal that the plugin is ready.
+ */
+export const moduleRoutes: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  fastify.post(
+    '/module-completed',
+    {
+      schema: schemas.moduleCompleted,
+      errorHandler(error, request, reply) {
+        if (error.validation) {
+          void reply.code(400);
+          return {
+            type: 'error',
+            message: 'That does not appear to be a valid module submission.'
+          };
+        } else {
+          fastify.errorHandler(error, request, reply);
+        }
+      }
+    },
+    async req => {
+      const { id } = req.body;
```

Comment:
Very nitpicky, but I would rename this to `moduleId`, so that it carries its context with it.